core/chains/evm/headtracker/head_tracker.go

```diff
@@ -79,22 +123,23 @@ func NewHeadTracker(
 		headListener:    NewEvmHeadListener(lggr, ethClient, config, chStop),
 		headSaver:       headSaver,
 		mailMon:         mailMon,
+		getNilHead:      func() *evmtypes.Head { return nil },
 	}
 }
 
 // Start starts HeadTracker service.
-func (ht *headTracker) Start(ctx context.Context) error {
+func (ht *headTracker[HTH, S, ID, BLOCK_HASH]) Start(ctx context.Context) error {
 	return ht.StartOnce("HeadTracker", func() error {
-		ht.log.Debugf("Starting HeadTracker with chain id: %v", ht.chainID.Int64())
+		ht.log.Debugf("Starting HeadTracker with chain id: %v", ht.chainID)
```

Comment:
since `chainID` is already a fmt.Stringer type, `Debugf` would call the `String()`.

Reply:
FYI: a common practice is to use `Debugw`, e.g.:
`ht.log.Debugw("Starting HeadTracker", "chainID", ht.chainID)`